src/main/java/org/sonar/plugins/coverage/generic/ReportParser.java

```diff
@@ -311,6 +311,9 @@ private void saveCoverageMeasure() {
   private void saveUnitTestMeasures() {
     for (Map.Entry<InputFile, UnitTestMeasuresBuilder> entry : unitTestMeasures.entrySet()) {
       InputFile inputFile = entry.getKey();
+      if (inputFile.type() == InputFile.Type.MAIN) {
+        throw new IllegalStateException("Unit tests report must be associated to test files, unexpected main file received: " + inputFile.absolutePath());
```

Comment:
@mpaladin Suggested message: "Line x of unit test result report /path/of/the/report refers to a file which is not configured as a test file: path/as/it/appears/in/the/report"